### PR TITLE
feat(port): add tabby

### PR DESCRIPTION
**Pull request by asciiden**
https://tabby.sh/static/32bf32ff6c87e8d18932.svg

tabby-image

### CODE REVIEW

**Comment by daltonmenezes** [comment]
Hi @asciiden , thanks man. About the screenshot, try to remote the opacity on the background if it's exists and should be awesome if you make it with a good zoom [like this one](https://github.com/daltonmenezes/aura-theme/pull/42#issuecomment-937362814)

I'll try to review your PR as soon as possible. 💜

File: src/ports/tabby/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const version = '1.0.0'
+  const templateFolder = resolve(__dirname, 'templates')
+
+  const background_color = '#21202e'
+
+  await createPort({
+    template: resolve(templateFolder, `tabby-colors.yaml`),
+    replacements: {
+      ...colorSchemes.dark,
+      background_color,
```
[the remark this answers is not in the document]
Comment (asciiden) [reply src/ports/tabby/index.ts]: hey, I have updated these in my latest push

File: src/ports/tabby/templates/tabby-colors.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+terminal:
+  searchOptions: {}
+  colorScheme:
+    name: {{ theme_name }}
```
[the remark this answers is not in the document]
Comment (asciiden) [reply src/ports/tabby/templates/tabby-colors.yaml:4]: theme_name restored in latest push

File: src/ports/tabby/templates/tabby-colors.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    foreground: '{{ accent7 }}'
+    background: '{{ accent12 }}'
+    selection: null
```
[the remark this answers is not in the document]
Comment (asciiden) [reply src/ports/tabby/templates/tabby-colors.yaml]: Not sure what selection color does... I checked the hex value and yet my eye can't see it. It's not in the manual config (GUI) as well. Changed it, nevertheless as it does not affect the colors of the terminal.

File: src/ports/tabby/templates/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  <tbody>
+    <tr>
+      <td><a href="https://github.com/asciiden">Dennis/ASCIIDEN</a></td>
+      {{{ author-tbody }}}
+    </tr>
+  </tbody>
+</table>
+
+{{{ footer }}}
```
Comment (daltonmenezes) [inline src/ports/tabby/templates/README.md:33]: Update it to:
```
{{{ basic-heading }}}

# Installation
1. Copy the contents of the [tabby-colors.yaml](https://github.com/{{ author.username }}/{{ slug }}/blob/main/packages/tabby/tabby-colors.yaml?raw=true) file.
2. Start {{ portName }}.
3. Click on the settings icon and click on `Config File`.
4. Scroll to the `terminal` section of the manifest and select everything from `terminal` to `background` (included).
5. Paste your copied content from the `tabby-colors.yaml` file.
6. Change the profile line to match your current profile (e.g. `profile: local:wsl-Debian`).
7. Save your file.

{{{ done }}}

# Contributors
<table>
  <thead>
    <tr>
      <td valign="bottom"><p align="center">
        <a href="https://github.com/asciiden">
          <img src="https://github.com/asciiden.png?size=100" align="center" />
        </a>
      </p></td>
      {{{ author-thead }}}
    </tr>
  </thead>

  <tbody>
    <tr>
      <td><a href="https://github.com/asciiden">Dennis/ASCIIDEN</a></td>
      {{{ author-tbody }}}
    </tr>
  </tbody>
</table>

{{{ footer }}}
```

File: src/ports/tabby/templates/tabby-colors.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+terminal:
+  searchOptions: {}
+  colorScheme:
+    name: {{ theme_name }}
+    foreground: '{{ accent7 }}'
+    background: '{{ accent12 }}'
+    selection: '{{ accent20 }}'
```
Comment (daltonmenezes) [inline src/ports/tabby/templates/tabby-colors.yaml:7]: Have you tested on Tabby after this change of selection to accent20? Worked well?

**Comment by daltonmenezes** [comment]
Hi @asciiden , almost done. After fixing the requested changes, don't forget to update the [main README](https://github.com/daltonmenezes/aura-theme/blob/main/README.md) by adding the Tabby icon at the end of the related category list.

You should use this following link as the Tabby icon:
```
https://github.com/daltonmenezes/assets/blob/master/images/icons/tabby.png?raw=true
```
thanks!

**Comment by asciiden** [comment]
Hey @daltonmenezes, I've updated the readme as well. Not sure if you got a notification for this so I'm adding a comment as well.

Thanks for the help throughout the PR.

**Comment by daltonmenezes** [comment]
Hi @asciiden , I'm really glad for your help, thanks. I was waiting for the last requested review changes, but I think you miss it. Check out [Here](https://github.com/daltonmenezes/aura-theme/pull/64#pullrequestreview-796022582) please.

**Comment by asciiden** [comment]
Hey, latest [push](https://github.com/daltonmenezes/aura-theme/pull/64/commits/a95133c340047831e50df4a226b945b266c79e45) - changed the theme_name to displayName. The theme name should not be in brackets:

image

File: src/ports/tabby/index.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { AuraAPI } from 'core'
+import { resolve } from 'path'
+
+export async function TabbyPort(Aura: AuraAPI) {
+  const { createPort, createReadme, colorSchemes } = Aura
+
+  const portName = 'Tabby'
+  const version = '1.0.0'
+  const displayName = 'Aura Theme'
```
Comment (daltonmenezes) [inline src/ports/tabby/index.ts:9]: This value should come from Aura constants info, I sent you in the review the full content of this file to override it.

File: src/ports/tabby/templates/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{{{ basic-heading }}}
+
+# Installation
+1. Copy the contents of the `tabby-colors.yaml` file.
```
Comment (daltonmenezes) [inline src/ports/tabby/templates/README.md:4]: Needs to update it to the content I sent you on the last review

**Comment by asciiden** [comment]
Not sure how to reply to a comment in a review...

accent20 for selection is fine but I can see that the second accent6 is breaking the terminal:

image

Changing it to accent12 (as background) to fix this.

**Comment by daltonmenezes** [comment]
Congrats @asciiden and thank you so much for your contribution 💜

**Comment by alexmk92** [comment]
This is awesome, just started using Tabby today and was so thankful to find someone's already made a port...thank you @asciiden!

**Comment by asciiden** [comment]
> This is awesome, just started using Tabby today and was so thankful to find someone's already made a port...thank you @asciiden!

**Dude, this is sick.** As this is my first official open-source contribution, it's ultra cool that it reached somebody! Thank YOU for your comment! Enjoy the port =)

**Pull request merged**